SImplify "astro add" by removing confusing multi-select

FredKSchott opened the pull request:
## Changes

- Our "astro add" multi-select had been causing users trouble.
- This change solves the usability issues by just removing the multi-select flow.
- The multi-select was a neat idea, but this new static output contains all the same information without the usability issues. Plus, it teaches you how to use the real CLI command.
- The analogy that I have come to think of this as: "npm doesn't have a multi-select for 'npm install', and they do just fine, so why should we?" 

**Bonus!** `create-astro` also updated with a few style changes and bug fixes.
**Bonus!** Added link to our full integration catalog.  
**Bonus!** Cleaned up the `printHelp()` function to be a bit more pretty across all the commands, and updated its API to be a bit more flexible.

### Screenshot: `astro add` multi-select replaced with this output
<img width="632" alt="Screen Shot 2022-06-25 at 11 33 16 PM" src="https://user-images.githubusercontent.com/622227/175802567-097dd515-e33f-4036-b865-66c7347a7f2e.png">

### Recording: The new, updated `create-astro` flow

https://user-images.githubusercontent.com/622227/175868477-ea018d75-3109-4966-ae69-20a2ff5464a6.mov


## Testing

- Tested manually (`astro add` isn't tested yet)
- This mainly removes logic, so bug risk should be lower than normal

## Docs

- Docs PR created: https://github.com/withastro/docs/pull/867

hippotastic commented:
Does the command specifying multiple integrations to add in one call? If so, that would be great, and it should be reflected in the help output somehow.

So rather than `astro add [integration]` we could write something like `astro add [integration names]`, `[integrations...]` or anything else that you see fit.

What I also currently find a bit misleading is that you're introducing astro add in the usage line as a command to install `[integrations]`, but then provide two lists: One of `UI Frameworks` (huh?), and one of `Integrations`. New developers could be confused if UI Frameworks are something else than integrations.

I'd personally also be confused why those integrations are all "recommended". Does that mean I should install all of them?

I'd therefore propose to change the list headers to something like `Example: Add UI Framework Support` and `Example: Add Other Popular Integrations`.

delucis commented:
Hippo's ideas seem smart — plus one on those.

Re: docs — yes we describe the current CLI output in detail in our install guide, which will need updating.

FredKSchott commented:
Agreed! Updated copy based on @hippotastic's comments.

> Re: docs — yes we describe the current CLI output in detail [in our install guide](https://docs.astro.build/en/install/auto/#install-any-official-astro-integrations-optional), which will need updating.

Great! PR to update created: https://github.com/withastro/docs/pull/867

> In fact, how will this work as part of create-astro? Do we just remove the add step and recommend running it when the setup CLI is done?

Good call out! I updated the PR to update create-astro for this change. Video preview added to OP.

delucis commented on packages/astro/src/core/logger/node.ts:
```diff
@@ -126,6 +126,7 @@ export const logger = {
 
 export function enableVerboseLogging() {
 	//debugPackage.enable('*,-babel');
+	console.log('ah!');
```
Was this left over by accident?

```suggestion
```

FredKSchott replied:
ah!

delucis commented on packages/create-astro/src/index.ts:
```diff
@@ -233,30 +211,40 @@ export async function main() {
+	await logAndWait('');
+	await logAndWait(`Stuck? Come join us at ${bold(cyan('https://astro.build/chat'))}`, 1000);
+	await logAndWait(dim('Good luck out there, astronaut.'));
+	await logAndWait('', 300);
```
Nice touch with the staggered output!

FredKSchott replied:
Thanks! I was kind of just playing around at first but it feels a lot more human.

The `1000` wait before "good luck out there" is a bit of a cheesy touch, so if anyone has any thoughts I'm okay to scrap it or shrink it down to a smaller wait. You can see it in action in the video above.

delucis commented on packages/create-astro/test/install-step.test.js:
```diff
@@ -26,6 +26,7 @@ describe('[create-astro] install', function () {
 			const seen = new Set();
 			const installPrompt = PROMPT_MESSAGES.install(FAKE_PACKAGE_MANAGER);
 			stdout.on('data', (chunk) => {
+				console.log(chunk.toString());
```
Looks like we need to disable ESLint for this line:

```suggestion
				// eslint-disable-next-line no-console
				console.log(chunk.toString());
```

delucis replied:
Or was it just for debugging and should be removed?

```suggestion
```

delucis reviewed (commented):
I’m a big plus one on these changes. The multi-select was also really hard to describe in docs, so this will make our life easier 😅

Spotted a couple of tiny things, but otherwise LGTM!

natemoo-re commented on .changeset/perfect-drinks-fold.md:
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Update the help output to improve formatting
```
These changelogs can be combined

FredKSchott replied:
They're both individually different features impacting different packages (for example, `create-astro` didn't receive a change to "update the help output"). 

I don't feel too strongly, but given that this is more correct I'd argue to keep as-is.

natemoo-re commented on packages/create-astro/src/index.ts:
```diff
@@ -64,7 +72,7 @@ export async function main() {
 			type: 'text',
 			name: 'directory',
-			message: 'Where would you like to create your app?',
+			message: 'Where would you like to create your new project?',
```
👍🏻

natemoo-re commented on packages/create-astro/src/index.ts:
```diff
@@ -233,30 +211,40 @@ export async function main() {
-	}
-	console.log(`\nStuck? Come join us at ${bold(cyan('https://astro.build/chat'))}`);
+	await logAndWait(
+		`Add frameworks like ${bold(cyan('react'))} and ${bold(
```
I'm fine with that reasoning! Can leave as is.
